### PR TITLE
Fast Export using AzureProvider doesn't split by newline #2517

### PR DESCRIPTION
**Pull request by prb112**
FYI - I found these will scripting a demo.

Fast Export with AzureProvider misses resources since it doesn't
complete #2520

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

**Review by tbieste — APPROVED** [review]
LGTM

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java [hunk 1]
```diff
@@ -446,17 +445,23 @@ protected boolean isTxTimeExpired() {
...
+                uploadId = BulkDataUtils.startPartUpload(cosClient, cosBucketName, this.currentObjectName);
 
-            if (logger.isLoggable(Level.FINE)) {
-                logger.fine(logPrefix() + " Started new multi-part upload: '" + this.uploadId + "'");
+                if (logger.isLoggable(Level.FINE)) {
+                    logger.fine(logPrefix() + " Started new multi-part upload: '" + this.uploadId + "'");
+                }
+            } else if (provider instanceof AzureProvider) {
+                // Must be Azure
+                uploadId = "azure";
```
Comment (lmsurpre) [inline fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java:463]: For COS/S3 its used for multipart uploads I think (each part of the upload gets its own id), but how is it used for Azure?

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java [hunk 2]
```diff
@@ -481,9 +485,7 @@ private void uploadWhenReady() throws Exception {
     private void uploadPartToAzure() throws Exception {
         // Azure API: Part number must be an integer between 1 and 10000
         int currentObjectPartNumber = uploadedParts.size() + 1;
-        if (logger.isLoggable(Level.FINE)) {
-            logger.fine(logPrefix() + " Uploading part# " + currentObjectPartNumber + " ["+ ioBuffer.size() + " bytes] for uploadId '" + uploadId + "'");
-        }
+        logger.fine(() -> logPrefix() + " Uploading part# " + currentObjectPartNumber + " ["+ ioBuffer.size() + " bytes] for uploadId '" + uploadId + "'");
```
Comment (lmsurpre) [inline fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java:488]: uploadId will always be "azure" here right?
```suggestion
        logger.fine(() -> logPrefix() + " Uploading part# " + currentObjectPartNumber + " ["+ ioBuffer.size() + " bytes]");
```

Comment (lmsurpre) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java:488]: I guess it doesn't hurt to say "for uploadId 'azure'" though, so feel free to ignore

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java (hunk 1 above)
Comment (prb112) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java:463]: it's not, however there are multiple uploadId == null checks further  in the code

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java (hunk 2 above)
Comment (prb112) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java:488]: Right - always azure.

**Pull request merged**